examples/games/desktoy.rs

```diff
@@ -0,0 +1,389 @@
+    bevy_transform.translation = new_translation.extend(bevy_transform.translation.z);
+
+    // Add the cursor drag velocity in the opposite direction to each pupil.
+    // Remember pupils are using local coordinates to move. So when the Bevy logo moves right they need to move left to
+    // simulate inertia, otherwise they will move fixed to the parent.
+    for mut pupil in &mut q_pupils {
+        pupil.velocity -= drag_velocity;
+    }
+}
+
+/// Quit when the user right clicks the Bevy logo
+fn quit(
+    cursor_world_pos: Res<CursorWorldPos>,
+    mut app_exit: EventWriter<AppExit>,
+    q_bevy_logo: Query<&Transform, With<BevyLogo>>,
+) {
+    // If the cursor is not within the primary window skip this system
+    let Some(cursor_world_pos) = cursor_world_pos.0 else {
+        return;
+    };
+
+    // If the cursor is within the Bevy logo radius send the [`AppExit`] event to quit the app
+    let bevy_logo_transform = q_bevy_logo.single();
+    if (bevy_logo_transform.translation.truncate() - cursor_world_pos).length() < BEVY_LOGO_RADIUS {
+        app_exit.send(AppExit);
+    }
+}
+
+/// Enable transparency for the window and make it on top
+fn toggle_transparency(
+    mut commands: Commands,
+    mut window_transparency: ResMut<WindowTransparency>,
+    mut q_instructions_text: Query<&mut Visibility, With<InstructionsText>>,
+    mut q_primary_window: Query<&mut Window, With<PrimaryWindow>>,
+) {
+    // Toggle the window transparency resource
+    window_transparency.0 = !window_transparency.0;
+
+    // Show or hide the instructions text
+    for mut visibility in &mut q_instructions_text {
+        *visibility = if window_transparency.0 {
+            Visibility::Hidden
+        } else {
+            Visibility::Visible
+        };
+    }
+
+    // Remove the primary window's decorations (e.g. borders), make it always on top of other desktop windows, and set the clear color to transparent
+    // only if window transparency is enabled
+    let mut window = q_primary_window.single_mut();
+    let clear_color;
+    (window.decorations, window.window_level, clear_color) = if window_transparency.0 {
+        (false, WindowLevel::AlwaysOnTop, Color::NONE)
+    } else {
+        (true, WindowLevel::Normal, Color::srgb(0.2, 0.2, 0.2))
+    };
+
+    // Set the clear color
+    commands.insert_resource(ClearColor(clear_color));
+}
+
+/// Move the pupils and bound them around
+fn move_pupils(time: Res<Time>, mut q_pupils: Query<(&mut Pupil, &mut Transform)>) {
+    for (mut pupil, mut transform) in &mut q_pupils {
+        // The wiggle radius is how much the pupil can move within the eye
+        let wiggle_radius = pupil.eye_radius - pupil.pupil_radius;
+        // Store the Z component
+        let z = transform.translation.z;
+        // Truncate the Z component to make the calculations be on [`Vec2`]
+        let mut translation = transform.translation.truncate();
+        // Decay the pupil velocity
+        pupil.velocity *= 0.93;
```

Comment:
The rate of decay seems frame rate dependent. At 120hz, the total duration of the bounce seems too short.

Reply:
Corrected the decay formula and made it frame rate independent.